src/PrestaShopBundle/Controller/Api/TranslationController.php

```diff
@@ -137,32 +120,23 @@ public function listTreeAction(Request $request)
 
             $search = $request->query->get('search');
 
-            if (in_array($type, ['modules', 'themes']) && empty($selected)) {
-                throw new Exception('This \'selected\' param is not valid.');
+            if (!in_array($type, ProviderDefinitionInterface::ALLOWED_TYPES)) {
+                throw new Exception(sprintf("The 'type' parameter '%s' is not valid", $type));
             }
 
-            switch ($type) {
-                case 'themes':
-                    $tree = $this->getNormalTree($lang, $type, $selected, $search);
-                    break;
-
-                case 'modules':
-                    $tree = $this->getModulesTree($lang, $selected, $search);
-                    break;
-
-                case 'mails':
-                    $tree = $this->getMailsSubjectTree($lang, $search);
-                    break;
-
-                case 'mails_body':
-                    $tree = $this->getMailsBodyTree($lang, $search);
-                    break;
+            if (ProviderDefinitionInterface::TYPE_THEMES === $type && '0' === $selected) {
```

Comment:
What is `'0'` in this case? Sounds like a good constant candidate

Reply:
The choice '0' is for Core translations

Yeah, can be in a constant